subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/BuildScopeScriptClassCompiler.java

```diff
@@ -15,40 +15,27 @@
  */
 package org.gradle.groovy.scripts.internal;
 
-import com.google.common.collect.Maps;
 import groovy.lang.Script;
 import org.codehaus.groovy.ast.ClassNode;
 import org.gradle.api.Action;
 import org.gradle.api.internal.initialization.loadercache.ClassLoaderId;
 import org.gradle.groovy.scripts.ScriptSource;
-import org.gradle.internal.Cast;
-
-import java.util.Map;
 
 /**
- * This in-memory cache is responsible for caching compiled build scripts during a build session.
- * If the compiled script is not found in this cache, it will try to find it in the global cache,
- * which will use the delegate script class compiler in case of a miss.
+ * ScriptClassCompiler that uses the cross-build in-memory cache and delegates to the given ScriptClassCompiler.
  */
-public class BuildScopeInMemoryCachingScriptClassCompiler implements ScriptClassCompiler {
+public class BuildScopeScriptClassCompiler implements ScriptClassCompiler {
```

Comment:
⭕️ This no longer needs to be build scoped. It can just be the "CachingScriptClassCompiler". You can probably squash this class and CrossBuildInMemoryCachingScriptClassCache into one.

Reply:
The delegate `ScriptClassCompiler` here is the `FileCacheBackedScriptClassCompiler` that is itself build scoped. Changing this may drag a lot of other moves. Let's keep this PR small and cleanup afterwards.

Reply:
Interesting, I wonder why that would be build scoped. But I agree, let's keep it simple.